packages/x-components/src/x-modules/search-box/components/search-input-placeholder.vue

```diff
@@ -0,0 +1,51 @@
+<template>
+  <label ref="placeholder" class="placeholder"></label>
+</template>
+
+<script lang="ts">
+  import Vue from 'vue';
+  import { Component, Prop } from 'vue-property-decorator';
+  import { State } from '../../../components/decorators/store.decorators';
+  import { XOn } from '../../../components/decorators/bus.decorators';
+  import { xComponentMixin } from '../../../components/x-component.mixin';
+  import { searchBoxXModule } from '../x-module';
+  import SearchInput from './search-input.vue';
+  @Component({
+    components: { SearchInput },
+    mixins: [xComponentMixin(searchBoxXModule)]
+  })
+  export default class SearchInputPlaceholder extends Vue {
+    @State('searchBox', 'query')
+    public query!: string;
```

Comment:
You can get access to the current query through the $x object.  You can see what's available [here](https://github.com/empathyco/x/blob/main/packages/x-components/src/plugins/x-bus.types.ts)

So you don't need to listen to the state directly